sql27thsep/sql27th sep.txt

```diff
@@ -0,0 +1,224 @@
+select substring(productCode,1,3) from Products;
+
+5. Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+select right('abcdefghijkl',5);
+
+6.From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+  select right([product code],4) from products;
+
+7. Write a query to find the length of the string 'SQL Server Functions'.
+select len('SQL SERVER Functions');
+
+8. From a 'Customers' table, find customers whose names are longer than 20 characters.
+select customername from Customer where len(customername)>20; 
+
+9. Compare the results of character count and byte count for the string 'SQL Server' with a trailing space.
+select datalength('sql server ');     _____________:output =11(bytecount)
+select len('sql server '); ________________________:output=10(char count)
+
+
+10. Write a query to find the byte count of an empty string and explain the result.
+select datalength('');_________________________output=0 bcoz here nothing is to be stored 
+
+11. Find the position of 'Server' in the string 'Microsoft SQL Server'.
+select CHARINDEX('server','microsoft sql server');
+
+12. From an 'Emails' table, write a query to extract the domain name from email addresses.
+select substring(emailid,charindex('@',emailid),len(emailid)) from employees;
+
+13. Find the position of the first number in the string 'ABC123DEF456'.
+select PATINDEX('%[0-9]%','ABC123DEF456');
+
+14. Write a query to find all product names from a 'Products' table that contain a number.
+select productname from products where patindex('%[0-9]%',productname)>0;
+
+15. Join the strings 'SQL', 'Server', and '2022' with spaces between them.
+select CONCAT_WS('-','sql','server','2022');
+
+16. From 'Employees' table with 'FirstName' and 'LastName' columns, create a 'FullName' column.
+select CONCAT_WS(' ',firstname,lastname) as fullname from employees;
+
+17. Join the array ('SQL', 'Server', '2022') with a hyphen as the separator.
+select CONCAT_WS('-','sql','server','2022');
+
+18. From an 'Addresses' table, combine 'Street', 'City', 'State', and 'ZIP' columns into a single address string.
+select concat_ws(',',street,city,state,zip) from address;
+
+19. Change all occurrences of 'a' to 'e' in the string 'database management'.
+select replace('database management','a','e');
+
+20. From a 'Products' table, write a query to replace all spaces in product names with underscores.
+select replace(productname,' ','-') from products;
+
+21. Create a string of 10 asterisks (*).
+select replicate('*',10);
+
+22. Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+select right(replicate('0',10)+[product code],10) from products;
+
+23. Insert the string 'New ' at the beginning of 'York City'.
+select concat_ws(' ','new'+ 'York city');    
```

Comment:
Why 'new'+ 'York city' then concat_ws ?

Reply:
had used that as i had used the concat_ws in prev. questions so...by default used that itself since it was working fine.it would have been worked with concat('new','york') also.